packages/xo-server-transport-icinga2/src/index.js

```diff
@@ -0,0 +1,126 @@
+import createLogger from '@xen-orchestra/log'
+import httpRequestPlus from 'http-request-plus'
+
+// =============================================================================
+
+const log = createLogger('xo:xo-server:transport-icinga2')
+
+// =============================================================================
+
+export const configurationSchema = {
+  type: 'object',
+
+  properties: {
+    server: {
+      type: 'string',
+      description: 'The icinga2 server address',
+    },
+    port: {
+      type: 'integer',
+      description: 'The access port',
+    },
+    user: {
+      type: 'string',
+      description: 'The icinga2 server username',
+    },
+    password: {
+      type: 'string',
+      description: 'The icinga2 server password',
+    },
+    filter: {
+      type: 'string',
+      description: 'The filter to use',
+    },
+    rejectUnauthorized: {
+      type: 'boolean',
+      description: 'Reject unauthorized certificates',
+      default: true,
+    },
+  },
+  additionalProperties: false,
+  required: ['server', 'port', 'user', 'password', 'filter'],
+}
+
+// =============================================================================
+
+export const OK = 0
+export const WARNING = 1
+export const CRITICAL = 2
+export const UNKNOWN = 3
+
+// =============================================================================
+
+class XoServerIcinga2 {
+  constructor({ xo }) {
+    this._xo = xo
+  }
+
+  // ---------------------------------------------------------------------------
+
+  configure(configuration) {
+    const ENDPOINT = '/v1/actions/process-check-result'
+    this._url =
+      'https://' +
+      configuration.user +
+      ':' +
+      configuration.password +
+      '@' +
+      configuration.server +
+      ':' +
+      configuration.port +
+      ENDPOINT
```

Comment:
Don't build URLs by hand!

This will break if any of the params contain special characters.

Use [`url.format`](https://devdocs.io/node/url#url_url_format_urlobject) instead (you may need to add a comment to disable an ESLint warning:

```js
// eslint-disable-next-line node/no-deprecated-api
```

Reply:
Is there a non-deprecated api to format url?

Reply:
No, unfortunately: https://github.com/nodejs/node/issues/25099

Reply:
I didn't need the comment line.
My call to `url.format` was succesfully commited.